xline/src/server/lock_server.rs

```diff
@@ -26,54 +40,218 @@ pub(crate) struct LockServer {
+        let keys = match wrapper.request {
+            RequestWrapper::DeleteRangeRequest(ref req) => {
+                vec![KeyRange::new(req.key.as_slice(), "")]
             }
-            Request::RequestDeleteRange(ref req) => vec![KeyRange {
-                start: req.key.clone(),
-                end: req.range_end.clone(),
-            }],
-            Request::RequestTxn(ref req) => req
+            RequestWrapper::RangeRequest(ref req) => {
+                vec![KeyRange::new(req.key.as_slice(), req.range_end.as_slice())]
+            }
+            RequestWrapper::TxnRequest(ref req) => req
                 .compare
                 .iter()
-                .map(|cmp| KeyRange {
-                    start: cmp.key.clone(),
-                    end: cmp.range_end.clone(),
-                })
+                .map(|cmp| KeyRange::new(cmp.key.as_slice(), cmp.range_end.as_slice()))
                 .collect(),
+            _ => vec![],
         };
-        let request_op = RequestOp {
-            request: Some(request),
+        Command::new(keys, wrapper, propose_id)
+    }
+
+    /// Propose request and get result with fast/slow path
+    async fn propose<T>(
+        &self,
+        request: T,
+        token: Option<String>,
+        use_fast_path: bool,
+    ) -> Result<(CommandResponse, Option<SyncResponse>), tonic::Status>
+    where
+        T: Into<RequestWrapper>,
+    {
+        let wrapper = match token {
+            Some(token) => RequestWithToken::new_with_token(request.into(), token),
+            None => RequestWithToken::new(request.into()),
         };
-        let wrapper = RequestWithToken::new(request_op.into());
-        let cmd = Command::new(key_ranges, wrapper, propose_id);
-        self.client.propose_indexed(cmd.clone()).await
+        let propose_id = self.generate_propose_id();
+        let cmd = Self::command_from_request_wrapper(propose_id, wrapper);
+        if use_fast_path {
+            let cmd_res = self.client.propose(cmd).await.map_err(|err| {
+                if let ProposeError::ExecutionError(e) = err {
+                    tonic::Status::invalid_argument(e)
+                } else {
+                    panic!("propose err {err:?}")
+                }
+            })?;
+            Ok((cmd_res, None))
+        } else {
+            let (cmd_res, sync_res) = self.client.propose_indexed(cmd).await.map_err(|err| {
+                if let ProposeError::ExecutionError(e) = err {
+                    tonic::Status::invalid_argument(e)
+                } else {
+                    panic!("propose err {err:?}")
+                }
+            })?;
+            Ok((cmd_res, Some(sync_res)))
+        }
     }
 
-    /// Generate propose id
-    fn generate_propose_id(&self) -> ProposeId {
-        ProposeId::new(format!("{}-{}", self.name, Uuid::new_v4()))
+    /// Crate txn for try acquire lock
+    fn try_acquire_txn(prefix: &str, lease_id: i64) -> TxnRequest {
```

Comment:
Why do we name this function as `try_acquire_txn`? The prefix `try_` looks like it should return a `Result`. In addition, this function only constructs a `txn` object. It should not fail.